src/restconfig/src/main/java/org/geoserver/catalog/rest/StylePackageResource.java

```diff
@@ -0,0 +1,464 @@
+            out = new BufferedOutputStream(new FileOutputStream(sldFile));
+
+            byte[] sldContent = FileUtils.readFileToByteArray(uploadedSldFile);
+
+            out.write(sldContent);
+
+            // This produces the error described in http://jira.codehaus.org/browse/GEOS-3840
+            //InputStream is = new FileInputStream(uploadedSldFile);
+            //SLDFormat sldFormat = new SLDFormat();
+            //Style style = (Style) sldFormat.read(is);
+
+            //SLDFormat format2 = new SLDFormat(true);
+            //format2.toRepresentation(style).write(out);
+
+            out.flush();
+
+        } catch (IOException e) {
+            throw new RestletException("Error creating file", Status.SERVER_ERROR_INTERNAL, e);
+        } finally {
+            IOUtils.closeQuietly(out);
+        }
+    }
+    /**
+     * Save the image resources in the styles folder
+     *
+     * @param directory
+     * @throws java.io.IOException
+     */
+    private void saveImageResources(File directory, File stylesDir) throws IOException {
+        File[] imageFiles = retrieveImageFiles(directory);
+
+        for (int i = 0; i < imageFiles.length; i++) {
+            FileUtils.copyFileToDirectory(imageFiles[i],
+                    stylesDir);
+                    //catalog.getResourceLoader().findOrCreateDirectory("styles"));
+        }
+    }
+
+    /**
+     * Returns a list of image files in the given directory
+     *
+     * @param directory
+     * @return
+     */
+    private File[] retrieveImageFiles(File directory) {
+        File[] matchingFiles = directory.listFiles(new FilenameFilter() {
+            public boolean accept(File dir, String name) {
+                return (name.endsWith("svg") || name.endsWith("png") || name.endsWith("jpg"));
+            }
+        });
+
+        return matchingFiles;
+    }
+
+    /**
+     * Returns the sld file in the given directory. If no sld file, throws an exception
+     *
+     * @param directory
+     * @return
+     */
+    private File retrieveSldFile(File directory) {
+        File[] matchingFiles = directory.listFiles(new FilenameFilter() {
+            public boolean accept(File dir, String name) {
+                return name.endsWith("sld");
+            }
+        });
+
+        if (matchingFiles.length == 0) {
+            throw new RestletException("No sld file provided:", Status.CLIENT_ERROR_FORBIDDEN);
+        }
+
+        LOGGER.fine("handleFileUpload (sldFile): " + matchingFiles[0].getAbsolutePath());
+
+        return matchingFiles[0];
+    }
+
+
+    /**
+     * Parses the sld file.
+     *
+     * @param sldFile
+     * @return
+     */
+    private Style parseSld(File sldFile) {
+        Style style = null;
+        InputStream is = null;
+
+        try {
+            is = new FileInputStream(sldFile);
+
+            SLDParser parser
+                    = new SLDParser(CommonFactoryFinder.getStyleFactory(null), is);
```

Comment:
I feel like I reviewed this code snippet before - do we want to try supporting several parsers here. Or use StyleHandler to support different kinds of files?